pkg/apis/serving/v1alpha1/knativeserving_types.go

```diff
@@ -105,6 +105,10 @@ type KnativeServingSpec struct {
 	// Allows specification of HA control plane
 	// +optional
 	HighAvailability *HighAvailability `json:"high-availability,omitempty"`
+
+	// Override resource requirements; keyed by container name
+	// +optional
+	Resources map[string]corev1.ResourceRequirements `json:"resources,omitempty"`
```

Comment:
The API conventions discourage maps for fields like this: https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#lists-of-named-subobjects-preferred-over-maps

Reply:
Good call. A struct with a `container` field is much clearer than a string key.

Reply:
Done.